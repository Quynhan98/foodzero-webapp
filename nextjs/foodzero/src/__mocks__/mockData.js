export const AVATAR_MOCK =
  'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf'

export const PRICE_LIST_MOCK = {
  name: 'Deep Sea Snow White Cod Fillet',
  price: 20,
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ',
}

export const CARD_BLOG = {
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FFeature%20Image.webp?alt=media&token=766a34ea-32e0-40e7-90e6-299f9af22eb1',
  title: 'Fruit and vegetables and protection against diseases',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  category: 'Fashion',
  srcAvatar: AVATAR_MOCK,
  name: 'Julie Christie',
  date: '2022-12-06T23:29:35.000Z',
  numberOfComments: 2,
}

export const CARD_FEATURE = {
  title: 'Premium Quality',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FCircleIcon.webp?alt=media&token=1df373f3-27a3-45f6-82fa-d997c47c718f',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Neque congue arcu',
}

export const CARD_CATEGORY = {
  href: '/',
  category: 'Starters',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcategory.webp?alt=media&token=9f307d7c-b242-479d-943a-4ae672ac601e',
}

export const USERS_MOCK = [
  {
    id: 'gff1',
    email: 'foodzero@gmail.com',
    password: '123456789',
  },
]

export const QUOTE_MOCK = [
  {
    id: '1',
    authorName: 'John Doe',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf',
  },
  {
    id: '2',
    authorName: 'Julie Christie',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet1.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Funnamed.webp?alt=media&token=72862fa3-6567-4043-aab0-cc449b30e602',
  },
  {
    id: '3',
    authorName: 'Dianne Russell',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet2.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcute-avatar.webp?alt=media&token=7b6d918e-3dec-43cf-a4bd-70afdbdc63e0',
  },
]

export const OUR_MENU_MOCK = [
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
]

export const FEATURES_MOCK = [CARD_FEATURE, CARD_FEATURE, CARD_FEATURE]

export const BLOG_SECTION_MOCK = [CARD_BLOG, CARD_BLOG]

export const CATEGORY_SECTION_MOCK = [
  CARD_CATEGORY,
  CARD_CATEGORY,
  CARD_CATEGORY,
]
