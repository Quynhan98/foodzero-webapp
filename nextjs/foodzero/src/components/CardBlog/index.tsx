import Image from 'next/image'
import { memo } from 'react'
import { Box, BoxProps, Heading, Text } from '@chakra-ui/react'

// Component
import BlogAvatar from '@components/BlogAvatar'

interface CardBlogProps extends BoxProps {
  imageUrl: string
  title: string
  description: string
  category: string
  srcAvatar: string
  name: string
  date: string
  numberOfComments: number
}

const CardBlog = ({
  imageUrl,
  title,
  description,
  category,
  srcAvatar,
  name,
  date,
  numberOfComments,
  ...rest
}: CardBlogProps) => {
  return (
    <Box position="relative" {...rest}>
      <Box
        position="absolute"
        top="12px"
        right="12px"
        background="bronzeYellow"
        textAlign="center"
        padding={{ base: '5px 20px', md: '7px 53px' }}
      >
        <Heading
          as="h6"
          fontSize={{ base: 'xs', md: 'common' }}
          lineHeight={{ base: 'xs', md: 'sm' }}
          color="white"
        >
          {category}
        </Heading>
      </Box>
      <Image src={imageUrl} alt="blog wallpaper" width={792} height={792} />
      <Box
        padding={{ base: '30px 15px', md: '68px 75px' }}
        textAlign="center"
        margin="0 auto"
      >
        <BlogAvatar
          src={srcAvatar}
          name={name}
          date={date}
          numberOfComments={numberOfComments}
        />
        <Heading
          as="h4"
          size={{ base: 'extraSmall', md: 'base' }}
          variant="secondary"
          pb="18px"
          pt={{ base: '20px', md: '48px' }}
          textAlign="start"
        >
          {title}
        </Heading>
        <Box as="hr" color="black" borderTop="4px dotted black" />
        <Text
          size={{ base: 'small', md: 'default' }}
          pt="18px"
          textAlign="start"
        >
          {description}
        </Text>
      </Box>
    </Box>
  )
}

export default memo(CardBlog)
