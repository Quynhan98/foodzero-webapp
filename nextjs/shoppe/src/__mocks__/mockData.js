export const PRODUCT_ITEM = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: 'Sold out',
}
