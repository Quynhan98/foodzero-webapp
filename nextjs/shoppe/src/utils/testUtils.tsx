import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>{component}</ChakraProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
