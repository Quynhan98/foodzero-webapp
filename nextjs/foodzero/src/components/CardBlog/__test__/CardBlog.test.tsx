import { render, screen } from '@utils/testUtils'

// Mocks
import { CARD_BLOG } from '@mocks/mockData'

// Components
import CardBlog from '@components/CardBlog'

describe('CardBlog component', () => {
  const props = { ...CARD_BLOG }

  it('Renders CardBlog correctly', () => {
    const { container } = render(<CardBlog {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('Should render data correctly', () => {
    render(<CardBlog {...props} />)

    const title = screen.getByRole('heading', { level: 4 })
    const category = screen.getByRole('heading', { level: 6 })
    const name = screen.getByText(props.name)

    expect(title).toHaveTextContent(props.title)
    expect(category).toHaveTextContent(props.category)
    expect(name).toHaveTextContent(props.name)
  })
})
