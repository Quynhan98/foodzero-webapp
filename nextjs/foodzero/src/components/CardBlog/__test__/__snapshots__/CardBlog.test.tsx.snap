// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBlog component Renders CardBlog correctly 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <div
      class="css-1p10zp0"
    >
      <h6
        class="chakra-heading css-19hfhgc"
      >
        Fashion
      </h6>
    </div>
    <img
      alt="blog wallpaper"
      data-nimg="1"
      decoding="async"
      height="792"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fimage-blog-155e2.appspot.com%2Fo%2Ffoodzero%252FFeature%2520Image.webp%3Falt%3Dmedia%26token%3D766a34ea-32e0-40e7-90e6-299f9af22eb1&w=1920&q=75"
      srcset="/_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fimage-blog-155e2.appspot.com%2Fo%2Ffoodzero%252FFeature%2520Image.webp%3Falt%3Dmedia%26token%3D766a34ea-32e0-40e7-90e6-299f9af22eb1&w=828&q=75 1x, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fimage-blog-155e2.appspot.com%2Fo%2Ffoodzero%252FFeature%2520Image.webp%3Falt%3Dmedia%26token%3D766a34ea-32e0-40e7-90e6-299f9af22eb1&w=1920&q=75 2x"
      style="color: transparent;"
      width="792"
    />
    <div
      class="css-18kc6l"
    >
      <div
        class="css-70qvj9"
      >
        <span
          class="chakra-avatar css-fup3kj"
        >
          <div
            aria-label="Julie Christie"
            class="chakra-avatar__initials css-100o20p"
            role="img"
          >
            JC
          </div>
        </span>
        <ul
          class="css-91sxya"
          role="list"
        >
          <li
            class="css-18m884n"
          >
            Julie Christie
          </li>
          <li
            class="css-1bfqwby"
          >
            December 06, 2022
          </li>
          <li
            class="css-1bfqwby"
          >
            06:29 AM
          </li>
          <li
            class="css-1ru1j6w"
          >
            2
             comments
          </li>
        </ul>
      </div>
      <h4
        class="chakra-heading css-11p5r12"
      >
        Fruit and vegetables and protection against diseases
      </h4>
      <hr
        class="css-113h5p7"
      />
      <p
        class="chakra-text css-17arz7z"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
